### PR TITLE
Move posts-related selectors under state/posts

### PR DESCRIPTION
**Pull request by sgomes**
This PR moves all posts-related selectors under `state/posts/selectors`, since posts state has already been modularised.

For consistency, it also turns default exports into named exports.

All existing code is updated to use the new paths.

#### Changes proposed in this Pull Request

* Move all posts-related selectors under `state/posts/selectors`
* Rewrite selectors to use named exports rather than default exports
* Update all importers accordingly
* Minor linting-related changes

#### Testing instructions

There is no functionality change in this PR, so it should be enough to ensure that all unit tests pass, and that there are no build warnings related to missing exports or the like.

### CODE REVIEW

**Comment by sgomes** [comment]
Rebased to fix conflicts.

**Comment by sgomes** [comment]
Rebased again. Still looking for a review on this one.

File: client/my-sites/post-type-list/post-actions-ellipsis-menu/edit.jsx [hunk 1]
```diff
@@ -32,6 +32,7 @@ function PostActionsEllipsisMenuEdit( { translate, canEdit, status, editUrl, bum
 			onClick={ bumpStat }
 			icon="pencil"
 			onMouseOver={ preloadEditor }
+			onFocus={ preloadEditor }
```
Comment (scinos) [inline client/my-sites/post-type-list/post-actions-ellipsis-menu/edit.jsx:35]: Just nothing this seems unrelated to this PR (but a valid change!)

**Review by scinos — APPROVED** [review]
LGTM

File: client/my-sites/post-type-list/post-actions-ellipsis-menu/edit.jsx (hunk 1 above)
Comment (sgomes) [reply client/my-sites/post-type-list/post-actions-ellipsis-menu/edit.jsx:35]: The linter wasn't too happy about the lack of an `onFocus`, so I added it 🙂 It seemed like a safe enough improvement.

**Pull request merged**